client/api_payment.go

```diff
@@ -83,3 +93,62 @@ func (c *client) DisableRefund(ctx context.Context, paymentAddress string, txOpt
 	}
 	return tx.TxResponse.TxHash, nil
 }
+
+// ListUserPaymentAccounts list payment info by user address
+func (c *client) ListUserPaymentAccounts(ctx context.Context, opts types.ListUserPaymentAccountsOptions) (types.ListUserPaymentAccountsResult, error) {
+	params := url.Values{}
+	params.Set("user-payments", "")
+
+	account := opts.Account
+	if account == "" {
+		acc, err := c.GetDefaultAccount()
+		if err != nil {
+			log.Error().Msg(fmt.Sprintf("get default account failed %s", err.Error()))
+			return types.ListUserPaymentAccountsResult{}, err
+		}
+		account = acc.GetAddress().String()
+	}
+
+	reqMeta := requestMeta{
+		urlValues:     params,
+		contentSHA256: types.EmptyStringSHA256,
+		userAddress:   account,
+	}
+
+	sendOpt := sendOptions{
+		method:           http.MethodGet,
+		disableCloseBody: true,
+	}
+
+	endpoint, err := c.getEndpointByOpt(opts.EndPointOptions)
+	if err != nil {
+		log.Error().Msg(fmt.Sprintf("get endpoint by option failed %s", err.Error()))
+		return types.ListUserPaymentAccountsResult{}, err
+	}
+
+	resp, err := c.sendReq(ctx, reqMeta, &sendOpt, endpoint)
+	if err != nil {
+		log.Error().Msg("the list of user's payments failed: " + err.Error())
+		return types.ListUserPaymentAccountsResult{}, err
+	}
+	defer utils.CloseResponse(resp)
+
+	paymentAccounts := types.ListUserPaymentAccountsResult{}
+	// unmarshal the json content from response body
+	buf := new(strings.Builder)
+	_, err = io.Copy(buf, resp.Body)
+	if err != nil {
+		log.Error().Msg("the list of user's payments failed: " + err.Error())
+		return types.ListUserPaymentAccountsResult{}, err
+	}
+
+	bufStr := buf.String()
+	err = xml.Unmarshal([]byte(bufStr), &paymentAccounts)
+
+	// TODO(annie) remove tolerance for unmarshal err after structs got stabilized
+	if err != nil && paymentAccounts.StreamRecords == nil {
```

Comment:
should be err != nil || paymentAccounts.StreamRecords == nil ??

Reply:
fixed